spec/message.abnf

```diff
@@ -1,18 +1,22 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = text-start / text-escape / expression
 
-declaration = input-declaration / local-declaration
+pattern = *(text / expression)
```

Comment:
Suggestion: make more clear that `simple-message` differs from `pattern` only in leading character constraints stop by removing `text`.
```suggestion
pattern = *(text-char / text-escape / expression)
```
or maybe keep `text` and better align nonterminals with the [data model](https://github.com/unicode-org/message-format-wg/blob/85e16398b07a91d8b1a16eccdec474af4e6198cc/spec/syntax.md#pattern) by defining both `simple-message` and `pattern` as a sequence of "parts":
```abnf
simple-message    = [simple-start-part *pattern-part] ; a pattern that doesn't start with "."
simple-start-part = simple-start-text / expression
simple-start-text = (simple-start-char / text-escape) [text] ; text that doesn't start with "."
simple-start-char = %x0-2D         ; omit .
                  / %x2F-5B        ; omit \
                  / %x5D-7A        ; omit {
                  / %x7C           ; omit }
                  / %x7E-D7FF      ; omit surrogates
                  / %xE000-10FFFF

text      = 1*(text-char / text-escape)
text-char = simple-start-char / "."

pattern      = *pattern-part
pattern-part = text / expression
```

Reply:
I think I prefer what I suggested before: making dot special for unquoted patterns. I like the conceptual simplicity and having fewer overall productions:

```suggestion
message = simple-message / complex-message
; simple-messages are any unquoted pattern 
; unquoted patterns cannot start with dot
; the empty string is a valid message
simple-message = [(unquoted-start / expression) pattern]
unquoted-start = %x0-2D ; omit . 
               / %x2F-5B ; omit \ 
               / %x5D-7A ; omit { 
               / %x7C ; omit } 
               / %x7E-D7FF ; omit surrogates 
               / %xE000-10FFFF
```

Reply:
That fails to support a leading escape.
```diff
-simple-message = [(unquoted-start / expression) pattern]
+simple-message = [(unquoted-start / text-escape / expression) pattern]
```

Reply:
So altogether, we're looking at ABNF like one of the following:
**Data Model Analog**
```abnf
message = simple-message / complex-message

; simple-message is a pattern that doesn't start with dot
; the empty string is a valid message
simple-message    = [simple-start-part *pattern-part]
simple-start-part = simple-start-text / expression
simple-start-text = (simple-start-char / text-escape) [text]
simple-start-char = %x0-2D         ; omit .
                  / %x2F-5B        ; omit \
                  / %x5D-7A        ; omit {
                  / %x7C           ; omit }
                  / %x7E-D7FF      ; omit surrogates
                  / %xE000-10FFFF

; text is a sequence of characters, some represented using escapes
text      = 1*(text-char / text-escape)
text-char = simple-start-char / "."

; pattern is a possibly-empty sequence of text and expression parts
pattern      = *pattern-part
pattern-part = text / expression

complex-message = *(declaration [s]) complex-body
…
```

**Fewer Productions**
```abnf
message = simple-message / complex-message

; simple-message is a pattern that doesn't start with dot
; the empty string is a valid message
simple-message = [(simple-start-char / text-escape / expression) pattern]
simple-start-char = %x0-2D         ; omit .
                  / %x2F-5B        ; omit \
                  / %x5D-7A        ; omit {
                  / %x7C           ; omit }
                  / %x7E-D7FF      ; omit surrogates
                  / %xE000-10FFFF

; pattern is a possibly-empty sequence of characters, escapes, and expressions
pattern   = *(text-char / text-escape / expression)
text-char = simple-start-char / "."

complex-message = *(declaration [s]) complex-body
…
```

Reply:
Renamed `text-start` as `simple-start-char` and mostly followed Richard's last suggestion.